woocommerce/woocommerce-ios
# [Shipping Labels] Sort saved packages

## Description


- Since `Storage.WooShippingPackagesResponse` has `customPackages` and `savedPredefinedPackages` as normal `Set` we need to sort them to make sure they always show in same order in the UI.

## Testing information


1. Install and set up the Woo Shipping extension on your store.
2. Build and run the app with the revampedShippingLabelCreation feature flag enabled.
3. Create an order with the processing status and at least one physical product.
4. In your WooCommerce product settings on the web, change the weight and dimension units and save your changes.
5. On the Orders tab in the app, open the order you created.
6. In the order details, select "Create Shipping Label."
7.  Tap "Select a Package."
10. Switch to Saved tab and check that the packages are loaded
11. Pull to refresh and check that refreshing finishes and packages are loaded and show in the same order as before

## Screenshots


| Before (random order)      | After (sorted) |
| ----------- | ----------- |
| ![Simulator Screen Recording - iPhone 16 Pro - 2024-12-20 at 10 28 41](https://github.com/user-attachments/assets/40f33aa1-24e2-4a94-8fbb-314e9b1ba82d)      | ![Simulator Screen Recording - iPhone 16 Pro - 2024-12-20 at 10 26 34](https://github.com/user-attachments/assets/e65c3100-4404-4f66-ac05-83509e81aee5)       |

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
